ext/mbstring/mbstring.c

```diff
@@ -3520,54 +3557,49 @@ PHP_FUNCTION(mb_convert_variables)
 		from_encoding = *elist;
 	} else {
 		/* auto detect */
-		from_encoding = NULL;
-		identd = mbfl_encoding_detector_new(elist, elistsz, MBSTRG(strict_detection));
-		if (identd != NULL) {
-			n = 0;
-			while (n < argc) {
-				if (mb_recursive_encoder_detector_feed(identd, &args[n], &recursion_error)) {
-					break;
-				}
-				n++;
-			}
-			from_encoding = mbfl_encoding_detector_judge(identd);
-			mbfl_encoding_detector_delete(identd);
-			if (recursion_error) {
+		int num = 0;
+		for (int n = 0; n < argc; n++) {
+			zval *zv = &args[n];
+			num += mb_recursive_count_strings(zv);
+		}
+		char **val_list = (char**)ecalloc(num, sizeof(char *));
+		size_t *len_list = (size_t*)ecalloc(num, sizeof(size_t));
+		int i = 0;
+		for (int n = 0; n < argc; n++) {
```

Comment:
With the other change done, those ``int`` that get compared to ``argc`` should be changed to ``size_t`` too